### ISSUE TITLE
horizon: captive-core flags shouldn't be checked when captive core ingestion is disabled

### ISSUE THREAD

**Issue by 2opremio** [stellar/go#3507]
The following invocation of horizon shouldn't error:

```
export NETWORK_PASSPHRASE="Public Global Stellar Network ; September 2015"
export HISTORY_ARCHIVE_URLS="https://s3-eu-west-1.amazonaws.com/history.stellar.org/prd/core-live/core_live_001"
export DATABASE_URL="postgres://postgres:postgres@localhost:5432/horizon?sslmode=disable"
export ENABLE_CAPTIVE_CORE_INGESTION="false"
export INGEST_FAILED_TRANSACTIONS=true
export STELLAR_CORE_URL="http://localhost:11626"
export STELLAR_CORE_DATABASE_URL="postgres://postgres:postgres@localhost:5432/core?sslmode=disable"
export INGEST="true"
horizon ingest verify-range --from 1 --to 64 --verify-state

2021/03/27 19:50:11 Invalid config: stellar-core-binary-path = /Users/fons/stellar-core/src/stellar-core, but corresponding option captive-core-config-append-path is not configured
```

It only works after setting `CAPTIVE_CORE_CONFIG_APPEND_PATH` to a phony value (e.g. `CAPTIVE_CORE_CONFIG_APPEND_PATH=/dev/null`).

This is an ugly workaround the user shouldn't go through.

**Comment by 2opremio** [comment]
Related: #3481

**Comment by 2opremio** [comment]
I think this is a blocker for 2.1.0

**Comment by bartekn** [comment]
Fixed in #3513. Let's keep tracking config validation and tests in https://github.com/stellar/go/issues/3193.
